src/rust/engine/fs/store/src/lib.rs

```diff
@@ -586,10 +592,12 @@ impl Store {
     initial_lease: bool,
   ) -> Result<Digest, String> {
     let local = self.local.clone();
-    let digest = local
+    let bytes = directory.to_bytes();
+    let digest = Digest::of_bytes(&bytes);
+    local
       .store_bytes(
         EntryType::Directory,
-        None,
+        digest.hash,
         directory.to_bytes(),
```

Comment:
To avoid encoding the directory twice:
```suggestion
        bytes,
```